pkg/virt-handler/mount-manager/container-disk/mount.go

```diff
@@ -53,178 +46,29 @@ type Mounter interface {
 	Unmount(vmi *v1.VirtualMachineInstance) error
 }
 
-type vmiMountTargetEntry struct {
-	TargetFile string `json:"targetFile"`
-	SocketFile string `json:"socketFile"`
-}
-
-type vmiMountTargetRecord struct {
-	MountTargetEntries []vmiMountTargetEntry `json:"mountTargetEntries"`
-	UsesSafePaths      bool                  `json:"usesSafePaths"`
-}
-
-func NewMounter(isoDetector isolation.PodIsolationDetector, mountStateDir string, clusterConfig *virtconfig.ClusterConfig) Mounter {
+func NewMounter(isoDetector isolation.PodIsolationDetector, clusterConfig *virtconfig.ClusterConfig, mountRecorder mountutils.MountRecorder) Mounter {
 	return &mounter{
-		mountRecords:               make(map[types.UID]*vmiMountTargetRecord),
 		podIsolationDetector:       isoDetector,
-		mountStateDir:              mountStateDir,
+		mountRecorder:              mountRecorder,
 		suppressWarningTimeout:     1 * time.Minute,
 		socketPathGetter:           containerdisk.NewSocketPathGetter(""),
 		kernelBootSocketPathGetter: containerdisk.NewKernelBootSocketPathGetter(""),
 		clusterConfig:              clusterConfig,
 	}
 }
 
-func (m *mounter) deleteMountTargetRecord(vmi *v1.VirtualMachineInstance) error {
-	if string(vmi.UID) == "" {
-		return fmt.Errorf("unable to find container disk mounted directories for vmi without uid")
-	}
-
-	recordFile := filepath.Join(m.mountStateDir, string(vmi.UID))
-
-	exists, err := diskutils.FileExists(recordFile)
-	if err != nil {
-		return err
-	}
-
-	if exists {
-		record, err := m.getMountTargetRecord(vmi)
-		if err != nil {
-			return err
-		}
-
-		for _, target := range record.MountTargetEntries {
-			os.Remove(target.TargetFile)
-			os.Remove(target.SocketFile)
-		}
-
-		os.Remove(recordFile)
-	}
-
-	m.mountRecordsLock.Lock()
-	defer m.mountRecordsLock.Unlock()
-	delete(m.mountRecords, vmi.UID)
-
-	return nil
+func (m *mounter) addMountTargetRecord(vmi *v1.VirtualMachineInstance, record []mountutils.MountTargetEntry) error {
+	return m.mountRecorder.AddMountRecord(vmi, record)
```

Comment:
Nit: this method looks a bit redundant. I think `m.mountRecorder.AddMountRecord(vmi, record)` can be called directly.

Reply:
@vasiliy-ul it's a good suggestion. I removed the 2 functions